File: python/backspace.py
#!/usr/bin/env python3
""" An Attempt to solve the Backspace problem on Kattis """

import sys


def process_line(line):
    output = []

    for index in range(len(line)):
        if line[index] == "<":
            output.pop()
        else:
            output.append(line[index])
    return ''.join(output)


if __name__ == "__main__":
    data = list(sys.stdin.readline().rstrip())

    print(process_line(data))

File: src/bin/backspace.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();

    // Get the Input
    let data = stdin.lock().lines().next().unwrap().unwrap();
    let line = data.into_bytes();
    let mut output = vec!();
    for character in &line{
        // 60 is the < symbol
        if *character == 60 as u8{
            output.pop();
        } else {
            output.push(*character);
        }
    }
    let s = String::from_utf8(output).unwrap();
    println!("{}", s);
}
